openshift/backup.cj.yaml

```diff
@@ -0,0 +1,260 @@
+    required: false
+    value: "7"
+  - name: "WEEKLY_BACKUPS"
+    displayName: "Number of Weekly Backups to Retain"
+    description: "The number of weekly backup files to be retained.  Used for the `rolling` backup strategy."
+    required: false
+    value: "4"
+  - name: "MONTHLY_BACKUPS"
+    displayName: "Number of Monthly Backups to Retain"
+    description: "The number of monthly backup files to be retained.  Used for the `rolling` backup strategy."
+    required: false
+    value: "1"
+  - name: "SUCCESS_JOBS_HISTORY_LIMIT"
+    displayName: "Successful Job History Limit"
+    description: "The number of successful jobs that will be retained."
+    value: "5"
+    required: true
+  - name: "FAILED_JOBS_HISTORY_LIMIT"
+    displayName: "Failed Job History Limit"
+    description: "The number of failed jobs that will be retained."
+    value: "2"
+    required: true
+  - name: "JOB_BACKOFF_LIMIT"
+    displayName: "Job Backoff Limit"
+    description: "The number of attempts to try for a successful job outcome."
+    value: "0"
+    required: false
+  - name: "JOB_DEADLINE_SECONDS"
+    displayName: "Job deadline (seconds)"
+    description: "The maximum amount of time to let this job run."
+    value: "600"
+    required: false
+objects:
+  - apiVersion: v1
+    data:
+      BACKUP_STRATEGY: ${BACKUP_STRATEGY}
+      DAILY_BACKUPS: ${DAILY_BACKUPS}
+      DATABASE_SERVICE_NAME: ${TARGET}
+      DEFAULT_PORT: ${DEFAULT_PORT}
+      MONTHLY_BACKUPS: ${MONTHLY_BACKUPS}
+      NUM_BACKUPS: ${NUM_BACKUPS}
+      POSTGRESQL_DATABASE: ${DATABASE_NAME}
+      WEEKLY_BACKUPS: ${WEEKLY_BACKUPS}
+    kind: ConfigMap
+    metadata:
+      labels:
+        app: ${DATABASE_NAME}-backup
+        cronjob: ${DATABASE_NAME}-backup
+        template: "${JOB_NAME}-config-template"
+      name: ${TARGET}-backup
+  - apiVersion: v1
+    kind: PersistentVolumeClaim
+    metadata:
+      finalizers:
+      - kubernetes.io/pvc-protection
+      labels:
+        app: ${DATABASE_NAME}-backup
+        cronjob: ${DATABASE_NAME}-backup
+        template: "${JOB_NAME}-config-template"
+      name: ${TARGET}-backup
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          storage: ${PVC_SIZE}
+      storageClassName: ${PVC_CLASS}
+  - apiVersion: batch/v1beta1
+    kind: CronJob
+    metadata:
+      name: ${TARGET}-backup
+    spec:
+      concurrencyPolicy: Forbid
+      failedJobsHistoryLimit: ${{FAILED_JOBS_HISTORY_LIMIT}}
+      jobTemplate:
+        metadata:
+          creationTimestamp: null
+          labels:
+            app: ${DATABASE_NAME}-backup
+            cronjob: ${DATABASE_NAME}-backup
+            template: "${JOB_NAME}-config-template"
+        spec:
+          backoffLimit: ${{JOB_BACKOFF_LIMIT}}
+          template:
+            metadata:
+              creationTimestamp: null
+            spec:
+              activeDeadlineSeconds: ${{JOB_DEADLINE_SECONDS}}
```

Comment:
Double checking here, are these variables meant to have double brackets?  There are a couple in this file.

Reply:
Those are YAML variables required for properly handling integers.  They do look off though!